src/fiskaltrust.Launcher/Extensions/DataProtectionExtensions.cs

```diff
@@ -0,0 +1,213 @@
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+using System.Security.Cryptography;
+using System.Text;
+using System.Xml.Linq;
+using fiskaltrust.Launcher.Common.Helpers;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.DataProtection.KeyManagement;
+using Microsoft.AspNetCore.DataProtection.XmlEncryption;
+using Serilog;
+
+namespace fiskaltrust.Launcher.Extensions
+{
+
+    unsafe static partial class KeyUtils
+    {
+        private const Int64 KEYCTL = 250;
+
+        [LibraryImport("libc", SetLastError = true, EntryPoint = "syscall")]
+        private static partial Int64 keyctl_read(Int64 syscall, Int32 cmd, Int32 key, byte* buffer, UIntPtr buflen);
```

Comment:
Any specific reason for not using the aliases for the respective datatypes in this class? (`int`, `long`, ..)

Reply:
It helped me not to get confused with the similarly named but not necessary equal C datatypes on the other side ^^ for these things where the bit size of the type is more important than the value range I like to use those 🤷